src/ServiceDesk/CustomerService.php

```diff
@@ -43,12 +45,12 @@ public function __construct(ConfigurationInterface $configuration = null, Logger
      */
     public function createCustomer(string $email, string $displayName): UserInterface
     {
-        $data = json_encode(['email' => $email, 'displayName' => $displayName]);
+        $data = $this->encodeJson(['email' => $email, 'displayName' => $displayName]);
 
         $ret = $this->exec($this->uri, $data);
 
-        return $this->json_mapper->map(
-            json_decode($ret),
+        return $this->prepareJsonMapper()->map(
```

Comment:
So, as I mention in another comment, here is what I would do here:

Just abstract the whole operation to a separate method e.g. `hydrate(object $object): object`. This method would then use `prepareJsonMapper()` in the way that is necessary for it - or use something else entirely.

Reply:
:heavy_check_mark: